### DISCUSSION TITLE
Headless form submission

### DISCUSSION THREAD

**Discussion by lukasIO** [statamic/cms#5164] [Q&A]
With https://github.com/statamic/cms/pull/4115 being merged, I wondered if there is a documented way how to submit forms to statamic, when statmic is used as a headless cms?
thanks!

**Comment by edalzell** [comment]
sure, send the relevant data to the form endpoint.

**Reply by lukasIO** [reply]
thanks, are there any docs about what the form endpoint is? I couldn't find it.

**Reply by jackmcdade** [reply]
If you write {{ form:create }} tag, look at the URL it outputs and bam, there's your Endpoint :)

**Reply by lukasIO** [reply]
maybe it makes sense to include the form endpoint url in the graphql forms query for people who don't use the templating at all.

**Reply by marcdap** [reply]
Hi everyone. Sorry to rescue this conversation. I think it's an interesting topic. We are using Statamic as headless cms and creating forms dynamically using graphql. We are not using templating at all, and our frontend is totally decoupled and located on a different server. I think that having the URL endpoint returned on the graphql request would be the best option. Do you think it's possible?

**Reply by jasonvarga** [reply]
It's just `http://yoursite.com/!/forms/{handle}`. You can get the `handle` from graphql, then just pop it in the string.

**Reply by marcdap** [reply]
Thanks for your answer! My other concern is if it would be submittable, as the documentation says that `_token` and `_params` are required, and are not available throw GraphQL. I apologize for the inconvenience.

**Reply by jasonvarga** [reply]
Providing the form url through graphql wouldn't have made a difference there, though.

The token is the csrf token. Get it however makes sense to you. You could hit /cp/auth/token if you want.
The params would correspond to form tag parameters. You could submit `{ _redirect: '/success' }` for example. They are optional.

>  as the documentation says that _token and _params are required

Where?

**Reply by marcdap** [reply]
You are right. providing the URL is not gonna make any difference. The URL case was before knowing that it's as easy to use `http://yoursite.com/!/forms/{handle}`.

I was assuming this use case would be the same as [submitting the form via ajax](https://statamic.dev/forms#using-ajax).
